Homework/js/lesson-5-task-1-4.js

```diff
@@ -0,0 +1,263 @@
+    }
+
+    return users.filter((user) => {
+        const firstUser = user[0].toLowerCase();
+
+        for (const letter of letterUsers) {
+            if (firstUser.includes(letter)) {
+                return;
+            }
+        }
+
+        return user;
+
+    });
+
+};
+console.log(filterUserListFilter(["Alex", "alex", "Max", "Via", "Petr", "Alexandra"], ['a','p']));
+
+
+
+
+
+
+
+
+
+// 2)
+// for
+const allLetterFor = (letterArray) => {
+    if (!Array.isArray(letterArray)) {
+        return;
+    }
+
+    const newLetterArray = [];
+
+    for (let i = 0; i < letterArray.length; i++) {
+        const ONE_ELEMENT_OF_ARRAY = letterArray[i].toUpperCase();
+        if (letterArray[i] === ONE_ELEMENT_OF_ARRAY) {
+            newLetterArray.push(letterArray[i].toLowerCase());
+        } else {
+            newLetterArray.push(letterArray[i].toUpperCase());
+        }
+    }
+
+    return newLetterArray;
+}
+console.log(allLetterFor(['a', 'B', 'C', 'L', 'g']));
+
+
+// while
+const allLetterWhile = (letterArray) => {
+    if (!Array.isArray(letterArray)) {
+        return;
+    }
+
+    const newLetterArray = [];
+    let i = 0;
+    while (i < letterArray.length) {
+        const ONE_ELEMENT_OF_ARRAY = letterArray[i].toUpperCase();
+        if (letterArray[i] === ONE_ELEMENT_OF_ARRAY) {
+            newLetterArray.push(letterArray[i].toLowerCase());
+        } else {
+            newLetterArray.push(letterArray[i].toUpperCase());
+        }
+        i++
+    }
+
+    return newLetterArray;
+}
+console.log(allLetterWhile(['a', 'B', 'C', 'L', 'g', 'g', 'EEE']));
+
+
+// for of
+const allLetterForOf = (letterArray) => {
+    if (!Array.isArray(letterArray)) {
+        return;
+    }
+
+    const newLetterArray = [];
+    for (const letter of letterArray) {
+        const ONE_ELEMENT_OF_ARRAY = letter.toUpperCase();
+        if (letter === ONE_ELEMENT_OF_ARRAY) {
+            newLetterArray.push(letter.toLowerCase());
+        } else {
+            newLetterArray.push(letter.toUpperCase());
+        }
+    }
+
+    return newLetterArray;
+}
+console.log(allLetterForOf(['a', 'B', 'C', 'L', 'GGGGg', 'g', 'EEE']));
+
+
+// map
+const allLetterMap = (letterArray) => {
+    if (!Array.isArray(letterArray)) {
+        return;
+    }
+
+    const stringNumbersArray = letterArray.map((val) => {
+        const ONE_ELEMENT_OF_ARRAY = val.toUpperCase();
+        if (val === ONE_ELEMENT_OF_ARRAY) {
+            return (val.toLowerCase());
+        } else {
+            return (val.toUpperCase());
+        }
+    })
+
+    return stringNumbersArray;
+}
+console.log(allLetterMap(['a', 'B', 'C', 'LJHHHL', 'GGGGg', 'g', 'EEE']));
+
+
+
+
+
+
+
+// 3)
+const numbersArrayFunction = (value) => {
+    if (!Array.isArray(value)) {
+        return;
+    }
+
+    const stringNumbersArray = []
+    for (let i = 0; i < value.length; i++) {
+        stringNumbersArray.push(`Value: ${value[i]}, Index: ${value.indexOf(value[i])}, Squared: ${Math.pow(value[i], 2)}`);
```

Comment:
create a variable for value[i], just to avoid code duplicates

Reply:
Corrected